src/main/java/org/asamk/signal/manager/Manager.java

```diff
@@ -686,6 +687,34 @@ public void setContactName(String number, String name) {
         account.save();
     }
 
+    @Override
+    public void setContactBlocked(String number, boolean blocked) {
+        ContactInfo contact = account.getContactStore().getContact(number);
```

Comment:
The number should be canonicalized before retrieving the contact, using `Utils.canonicalizeNumber(number, username)`